### PR TITLE
add method to get voltage levels and their equipments

### PR DESCRIPTION
**Pull request by YenguiSeddik**

### CODE REVIEW

File: src/test/java/org/gridsuite/network/map/VoltageLevelsEquipmentsMapDataTest.java [hunk 1]
```diff
@@ -0,0 +1,13 @@
+package org.gridsuite.network.map;
```
Comment (FranckLecuyer) [inline src/test/java/org/gridsuite/network/map/VoltageLevelsEquipmentsMapDataTest.java:1]: Missing license and author

File: src/test/java/org/gridsuite/network/map/VoltageLevelConnectableMapDataTest.java [hunk 2]
```diff
@@ -0,0 +1,13 @@
+package org.gridsuite.network.map;
```
Comment (FranckLecuyer) [inline src/test/java/org/gridsuite/network/map/VoltageLevelConnectableMapDataTest.java:1]: Missing license and author

File: src/main/java/org/gridsuite/network/map/model/VoltageLevelsEquipmentsMapData.java [hunk 3]
```diff
@@ -0,0 +1,17 @@
+package org.gridsuite.network.map.model;
```
Comment (FranckLecuyer) [inline src/main/java/org/gridsuite/network/map/model/VoltageLevelsEquipmentsMapData.java:1]: Missing license and author

File: src/main/java/org/gridsuite/network/map/model/VoltageLevelConnectableMapData.java [hunk 4]
```diff
@@ -0,0 +1,17 @@
+package org.gridsuite.network.map.model;
```
Comment (FranckLecuyer) [inline src/main/java/org/gridsuite/network/map/model/VoltageLevelConnectableMapData.java:1]: Missing license and author

**Review by FranckLecuyer — CHANGES_REQUESTED** [review]

**Review by FranckLecuyer — APPROVED** [review]

File: src/main/java/org/gridsuite/network/map/NetworkMapService.java [hunk 5]
```diff
@@ -864,6 +872,19 @@ public List<VoltageLevelMapData> getVoltageLevels(UUID networkUuid, String varia
             substationsId.stream().flatMap(id -> network.getSubstation(id).getVoltageLevelStream().map(NetworkMapService::toMapData)).collect(Collectors.toList());
     }
 
+    public List<VoltageLevelsEquipmentsMapData> getVoltageLevelsAndConnectable(UUID networkUuid, String variantId, List<String> substationsId) {
+        Network network = getNetwork(networkUuid, substationsId == null ? PreloadingStrategy.COLLECTION : PreloadingStrategy.NONE, variantId);
+        List<VoltageLevel> voltageLevels =  substationsId == null ?
```
Comment (laurgarn) [inline src/main/java/org/gridsuite/network/map/NetworkMapService.java:877]: You could have kept the "Stream style" all the way, as in `getVoltageLevels` above
```
    public List<VoltageLevelsEquipmentsMapData> getVoltageLevelsAndConnectable(UUID networkUuid, String variantId, List<String> substationsId) {
        Network network = getNetwork(networkUuid,
            substationsId == null ? PreloadingStrategy.COLLECTION : PreloadingStrategy.NONE,
            variantId);
        Stream<VoltageLevel> voltageLevels = substationsId == null ?
            network.getVoltageLevelStream() :
            substationsId.stream().flatMap(id -> network.getSubstation(id).getVoltageLevelStream());

        return voltageLevels.map(vl -> VoltageLevelsEquipmentsMapData.builder()
                .voltageLevel(toMapData(vl))
                .equipments(vl.getConnectableStream().map(NetworkMapService::toMapData).collect(Collectors.toList()))
                .build())
            .collect(Collectors.toList());
    }
```

**Review by laurgarn — APPROVED** [review]

**Pull request merged**